game/src/contrib/systems/HealthSystem.java

```diff
@@ -34,22 +34,31 @@ public void execute() {
                 .map(this::applyDamage)
                 // Filter all dead entities
                 .filter(hsd -> hsd.hc.isDead())
-                .filter(
-                        hsd -> {
-                            if (!hsd.ac.hasAnimation(AdditionalAnimations.DIE)
-                                    || hsd.ac
-                                            .getAnimation(AdditionalAnimations.DIE)
-                                            .get()
-                                            .isLooping()) return true;
-                            if (!hsd.ac.isCurrentAnimation(AdditionalAnimations.DIE)) {
-                                hsd.ac.currentAnimation(AdditionalAnimations.DIE);
-                            }
-                            return hsd.ac.currentAnimation().isFinished();
-                        })
+                .filter(this::checkAnimationStatus)
```

Comment:
Ich weiß es war vorher bereits Filter, aber eigentlich ist Filter nur ein Entscheiden. ich würde dort keine Änderungen erwarten. Wie z.b. die Animation auszutauschen

Reply:
Da hast du vollkommen Recht
hab das in nem neuen Kommentar in #681 auch in Frage gestellt
War mir nur unsicher, ob so eine Diskussion im PR oder im Issue gestartet werden sollte ^^

Reply:
pr (am code) passt schon, ist ja ein implementierungsdetail. 

inhaltlich sehe ich das auch wie @AHeinisch, bei einer filter-operation würde ich keine nebeneffekte erwarten (wollen).

Reply:
mir scheint das auch irgendwie wird zu sein. 
In meinen kopf müsste der ablauf doch eh

mach schaden -> für alle die tot sind -> trigger onDeth -> für alle mit Todesanimation -> set Todesanimation

oder?

Reply:
> Soll das an dieser Stelle nur ein Safeguard sein?
Zumal die DeathAnimation dann in removeDeadEntities(), die nach diesem Filter erst aufgerufen wird, auch noch mal gesetzt wird
>Wäre das nicht sinniger z.B. in HealthComponent bei triggerOnDeath unabhängig vor dem call auf onDeath.accept() zu setzen?
Natürlich mit Check, ob die Entity ne DrawComponent und ne DeathAnimation hat

Für mich wirkt das bei triggerOnDeath am sinnvollsten

Reply:
Für mich auch, solange wir das "triggerOnDeath" im System meinen. 
Also irgendwas wie 
```
...java
...
hc.triggerOnDeath();
if(dc.hasAnimation(AdditionalAnimations.DIE) dc.currentAnimation(AdditionalAnimations.DIE)
...

Reply:
Da ich gerade bei dem UI versuche es sauber zu basteln. Vielleicht ist das sterben nicht wirklich im Healthsystem richtig.

Reply:
Wie du meinen?
Wo dann?

Reply:
Okay nochmal genau angeschaut:
Fall Tod
 - hat keine TodesAnimation
   - Wird entfernt 
 - aktuelle Animation ist nicht TodesAnimation
   - Ändere auf TodesAnimation
 - TodesAnimation ist looping
   - Wird entfernt 
 - TodesAnimation ist zu ende
   - Wird entfernt
 - Todesanimation ist nicht zu ende
   - Wird nicht entfernt
   
   
Okay die Zeile die micht gestört hat kann davor über ein Map gelöst werden.
```
.filter(hsd -> hsd.hc.isDead())
.map(hsd -> updateToDeathAnimation(hsd))
.filter(hsd -> heIsDead(hsd))
.forEach(this::removeDeadEntities);
```
updateToDeathAnimation versucht die Animation zu ändern, wenn aktuell nicht die Todesanimation ist.

heIsDead kontrolliert also nur noch isfinished | isLooping | Animation fehlt

Reply:
`heIsDead` sollte ehr `shouldBeRemoved`

Reply:
klingt doch irgendwie nach einem plan, oder? 

ich würde neben der anmerkung von @AMatutat obendrüber vielleicht auch eher `updateDeathAnimation` erwarten als `updateToDeathAnimation` (also ohne das "To" zwischendrin), aber das trifft es ja auch nicht so richtig. vielleicht `changeAnimationToDeathAnimation`? (krass, typisch java)

Reply:
Hab den Animationswechsel nu mal in triggerDeathAnimation() rausgezogen